### ISSUE TITLE
Problems with RuboCop::Cop::Layout::MultilineAssignmentLayout

### ISSUE THREAD

**Issue by sergey-alekseev** [rubocop-hq/rubocop#7431]
I enabled the `Layout/MultilineAssignmentLayout` cop (which is [disabled by default](https://github.com/rubocop-hq/rubocop/blob/ef25ea2016ec15745df013ecccf94896d7183dac/config/default.yml#L848)) and ran into several problems outlined below. So far I was impressed with Rubocop fixing layout offences using the autocorrect feature. However not with this cop 😿 

--------

## Expected behavior

`rubocop green.rb` not to be green actually. Copied from [green.rb](https://github.com/sergey-alekseev/rubocop-multiline-assignment-layout-example/blob/ef36ce02a0980cee69e647fc53d72c0d0d260d3b/green.rb):  
```ruby
# frozen_string_literal: true

@variable =
  begin
     end

@variable =
  begin
                    @something
  rescue StandardError
    false
                  end

# only w/ ", @variable2"
@variable1, @variable2 =
  @something.partition do |argument|
  argument > 1
end
```

## Actual behavior

It's green 😕 

## Steps to reproduce the problem

```
git clone https://github.com/sergey-alekseev/rubocop-multiline-assignment-layout-example
git reset --hard ef36ce02a0980cee69e647fc53d72c0d0d260d3b
rubocop green.rb
```

--------

## Expected behavior

`rubocop -a autocorrect.rb` to do it's job better. Copied from [autocorrect.rb](https://github.com/sergey-alekseev/rubocop-multiline-assignment-layout-example/blob/ef36ce02a0980cee69e647fc53d72c0d0d260d3b/autocorrect.rb):  
```ruby
# before `rubocop --safe-auto-correct autocorrect.rb`
@variable = if @conditional
              1
            else
              2
            end

# after `rubocop --safe-auto-correct autocorrect.rb`
# @variable =
#   if @conditional
#     1
#   else
#     2
#              end

# before `rubocop --safe-auto-correct autocorrect.rb`
@variable = begin
              1
            end

# after `rubocop --safe-auto-correct autocorrect.rb`
# @variable =
#   begin
#                1
#              end

# before `rubocop --safe-auto-correct autocorrect.rb`
@variable = begin
              @something
            rescue StandardError
              false
            end

# after `rubocop --safe-auto-correct autocorrect.rb`
# @variable =
#   begin
#                @something
#   rescue StandardError
#     false
#              end

# before `rubocop --safe-auto-correct autocorrect.rb`
@variable = begin
  if @conditional
    1
  else
    2
  end
end

# after `rubocop --safe-auto-correct autocorrect.rb`
# @variable =
#   begin
#    if @conditional
#      1
#    else
#      2
#    end
#  end

# before `rubocop --safe-auto-correct autocorrect.rb`
if @variable
  if @variable
    @variable = @conditional ?
      :a : :b
  end
end

# after `rubocop --safe-auto-correct autocorrect.rb`
# if @variable
#   if @variable
#     @variable =
#       @conditional ?
#            :a : :b
#   end
# end
```

## Actual behavior

It requires manually fixing 😕 

## Steps to reproduce the problem

```
git clone https://github.com/sergey-alekseev/rubocop-multiline-assignment-layout-example
git reset --hard ef36ce02a0980cee69e647fc53d72c0d0d260d3b
rubocop --safe-auto-correct autocorrect.rb
```

## RuboCop version

```
$ rubocop -V
0.75.0 (using Parser 2.6.5.0, running on ruby 2.5.5 x86_64-darwin18)
```
